spec/support/shared_examples/container.rb

```diff
@@ -185,6 +185,26 @@
           expect(container[:item].call).to eq('item')
         end
       end
+
+      context 'with option lazy_singleton: true' do
+        it 'registers and resolves a proc' do
+          container.register(:item, proc { 'item' }, lazy_singleton: true)
```

Comment:
Oops, just noticed I even ended up leaving one of my earlier names for this option here.